### ISSUE TITLE
Set output directory to be inside the projects directory 

### ISSUE THREAD

**Issue by KabelitzJ** [build2/build2#211]
I am new to build2 and are currently following along the toolchain introduction.
There, the build output (and i think the build configs later as well) are put in directories parallel to the project e.g.

```
demo/ <-- actual project 
demo-out/ <-- build output of the project above 
```

Is it currently possible to set the output path to be inside the actual projects directory?

```
demo/
- build/
  - ...
- out/
- src/
  - demo.cpp
  - buildfile
- buildfile
```

**Comment by boris-kolpackov** [comment]
No, this is intentionally not supported on purpose. The main reason is that `build2` is a "multi-repo first" toolchain and you would often build multiple unrelated projects in a shared build configuration (what you refer to as the output directory). You can read more on this here: https://github.com/build2/build2/issues/187

If you really want the output directory to be inside your "project" (i.e., what you clone), that what you can do is push the actual "build system project" into a subdirectory by making it a package:

```
bdep new --empty demo
cd demo
bdep new -t exe --package demo
tree ../demo/
../demo/
├── demo/
│   ├── build/
│   │   ├── bootstrap.build
│   │   └── root.build
│   ├── demo/
│   │   ├── buildfile
│   │   ├── demo.cxx
│   │   └── testscript
│   ├── buildfile
│   ├── manifest
│   └── README.md
├── buildfile
├── packages.manifest
├── README.md
└── repositories.manifest
```

Now you will be able to create the output directory as the subdirectory of top-level `demo/`.

**Comment by KabelitzJ** [comment]
@boris-kolpackov First of all, thank you for your fast and helpful answer. 
I am currently setting my project up like this:

I create my "main" project
`bdep new -t empty hello`

I create my libraries that my project uses internally
`bdep new --package -l c++ -t lib,split,no-tests,subdir=core,no-subdir-source hello/core`
`bdep new --package -l c++ -t lib,split,no-tests,subdir=math,no-subdir-source hello/math`

I create an executable that uses the libraries
`bdep new --package -l c++ -t exe,no-tests,subdir=src demo`

I create a project configuration
`bdep init -C ./out/gcc/debug/ cc @gcc_debug config.cxx=g++ config.cxx.coptions=-g`

I build the project
`b @gcc_debug`

And it works just fine.

But somehow the `demo.exe` does not only get put into `./out/gcc/debug/demo/src/` but also next to the `demo.cxx` in `demo/src/`. 

```
hello/
├── demo/
│   └── src/
│       ├── demo.cxx
│       └── demo.exe <-- here
└── out/
    └── gcc/
        └── debug/
            └── demo/
                └── src/
                    └── demo.exe <-- here
```

When i call `b clean` both binaries get removed and when I build again the binary is again in both places.

Am I doing something wrong with my configuration to cause this behavior?

**Comment by boris-kolpackov** [comment]
No, this is the expected (default) behavior: by default `bdep` configures your source tree to "forward" to the default (normally first creates) build configuration. This forwarding has two effects:

1. You are able to run `b` in the corresponding source directory in order to build in the forwarded-to configuration.

2. Some "interesting" build results (by default executables and documentation; but this is customizable in `buildfile`) are "backlinked" (normally symlinked) in the source directory.

The motivation for this functionality is that it makes it more convenient to develop by not having to jump between source and output trees.

But you can disable this if you would like by doing the following instead of `bdep init -C ...`:

```
bdep config create --no-forward ./out/gcc/debug/ cc @gcc_debug config.cxx=g++ config.cxx.coptions=-g
bdep init @gcc_debug
```

See [`bdep-config(1)`](https://build2.org/bdep/doc/bdep-config.xhtml) for details.
